go/private/tools/overlay_repository.bzl

```diff
@@ -0,0 +1,76 @@
+# Copyright 2018 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This file was copied from github.com/bazelbuild/bazel-gazelle.
+# We need a copy here to avoid a circular dependency in go_rules_dependencies.
+# Do not edit directly.
```

Comment:
Maybe a comment/issue number for a plan on how to fix this in the long term?

Reply:
I added some explanation on why the copy is needed. Until we have recursive workspaces though, we won't be able to load any external .bzl files from any file transitively loaded from WORKSPACE.